securedrop_client/api_jobs/downloads.py

```diff
@@ -191,41 +131,81 @@ def _check_file_integrity(cls, etag: str, file_path: str) -> bool:
         calculated_checksum = binascii.hexlify(hasher.digest()).decode('utf-8')
         return calculated_checksum == checksum
 
-    def _decrypt_file(
-        self,
-        session: Session,
-        db_object: Union[File, Message, Reply],
-        filepath: str,
-    ) -> None:
+
+class MessageDownloadJob(DownloadJob):
+    '''
+    Download and decrypt a message from a source.
+    '''
+
+    def __init__(self, uuid: str, data_dir: str, gpg: GpgHelper) -> None:
+        super().__init__(data_dir)
+        self.uuid = uuid
+        self.gpg = gpg
+
+    def get_db_object(self, session: Session) -> Message:
         '''
-        If a File: decrypt and save plaintext to a file on the filesystem.
-        If a Message or Reply: decrypt and save plaintext to local database.
+        Override DownloadJob.
         '''
-        try:
-            # Store plaintext in a file named after the downloaded file name, but without the
-            # extensions, e.g. 1-impractical_thing-doc.gz.gpg -> 1-impractical_thing-doc
-            fn_no_ext, _ = os.path.splitext(os.path.splitext(os.path.basename(filepath))[0])
-            plaintext_filepath = os.path.join(self.data_dir, fn_no_ext)
+        return session.query(Message).filter_by(uuid=self.uuid).one()
+
+    def call_download_api(self, api_client: API) -> Tuple[str, str]:
+        '''
+        Override DownloadJob.
+        '''
+        return api_client.download_submission(SdkSubmission(uuid=self.uuid))
```

Comment:
I think the submission object here also needs the `source_uuid`, I'm getting a 404 locally when trying to fetch submissions. Tbh this is an SDK bug, if we need the source UUID to fetch the submission (which from reading the source over there just now it appears that we do) then we should error when we initialize the `SdkSubmission` object. However in the meantime we'll need to pass the `source_uuid` to `SdkSubmission.__init__`.